src/freeze-dry/inline-styles.js

```diff
@@ -44,7 +44,7 @@ async function inlineLinkedStylesheets({rootElement, docUrl}) {
         } catch (err) {
             stylesheetText = '/* Oops! Freeze-dry failed to save this stylesheet. */'
         }
-        const styleEl = rootElement.ownerDocument.createElement('style')
+        const styleEl = rootElement.createElement('style')
```

Comment:
You have been mixing up documents and their `<html>` element (= document.documentElement). DOMParser returns a document, not the documentElement. Check and fix all assignments to rootElement and the problem should disappear.